cost_estimation_test.go

```diff
@@ -0,0 +1,65 @@
+package tfe
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCostEstimationsRead(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	rTest, rTestCleanup := createCostEstimatedRun(t, client)
+	defer rTestCleanup()
+
+	t.Run("when the costEstimation exists", func(t *testing.T) {
+		p, err := client.CostEstimations.Read(ctx, rTest.CostEstimation.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, p.LogReadURL)
+		assert.Equal(t, p.Status, CostEstimationFinished)
+		assert.NotEmpty(t, p.StatusTimestamps)
+	})
+
+	t.Run("when the costEstimation does not exist", func(t *testing.T) {
+		p, err := client.CostEstimations.Read(ctx, "nonexisting")
+		assert.Nil(t, p)
+		assert.Equal(t, err, ErrResourceNotFound)
+	})
+
+	t.Run("with invalid costEstimation ID", func(t *testing.T) {
+		p, err := client.CostEstimations.Read(ctx, badIdentifier)
+		assert.Nil(t, p)
+		assert.EqualError(t, err, "invalid value for costEstimation ID")
+	})
+}
+
+func TestCostEstimationsLogs(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	rTest, rTestCleanup := createCostEstimatedRun(t, client)
+	defer rTestCleanup()
+
+	t.Run("when the log exists", func(t *testing.T) {
+		p, err := client.CostEstimations.Read(ctx, rTest.CostEstimation.ID)
+		require.NoError(t, err)
+
+		logReader, err := client.CostEstimations.Logs(ctx, p.ID)
+		require.NoError(t, err)
+
+		logs, err := ioutil.ReadAll(logReader)
+		require.NoError(t, err)
+
+		assert.Contains(t, string(logs), "1 to add, 0 to change, 0 to destroy")
```

Comment:
We should check for for something cost estimation specific here, to make sure we have received cost estimation output as well.

Reply:
You're right. What's here is kinda dumb.

Reply:
I don't think we want to test for anything specific yet since it will be changing. Checking for read errors is good, though.